packages/core/http/core-http-server-internal/src/lifecycle/on_post_auth.ts

```diff
@@ -49,13 +57,26 @@ export function adoptToHapiOnPostAuthFormat(fn: OnPostAuthHandler, log: Logger)
     const hapiResponseAdapter = new HapiResponseAdapter(responseToolkit);
     try {
       const result = await fn(CoreKibanaRequest.from(request), lifecycleResponseFactory, toolkit);
+
       if (isKibanaResponse(result)) {
         return hapiResponseAdapter.handle(result);
       }
+
       if (postAuthResult.isNext(result)) {
         return responseToolkit.continue;
       }
 
+      if (postAuthResult.isAuthzResult(result)) {
+        Object.defineProperty(request.app, 'authzResult', {
```

Comment:
The object itself is just `Record<string, boolean>`. In terms of performance for the doing it for every request:
```yaml
Object.freeze (small object) x 50,090,673 ops/sec ±0.09% (99 runs sampled)
Normal assignment (small object) x 1,861,625,748 ops/sec ±0.50% (98 runs sampled)
Object.freeze (large object) x 541,953 ops/sec ±0.69% (92 runs sampled)
Normal assignment (large object) x 3,993 ops/sec ±0.43% (98 runs sampled)
Fastest is Normal assignment (small object)
```
We might measure latency for the routes once we switch a couple of them to the new security config. Though it shouldn't have any performance impact.


The benchmark for the reference:

```js
const Benchmark = require('benchmark');

const suite = new Benchmark.Suite();

const smallObject = { a: 1, b: 2, c: 3 };
const largeObject = {};
for (let i = 0; i < 1000; i++) {
  largeObject[`key${i}`] = `value${i}`;
}

suite
  .add('Object.freeze (small object)', function () {
    Object.freeze({ a: 1, b: 2, c: 3 });
  })
  .add('Normal assignment (small object)', function () {
    const obj = { a: 1, b: 2, c: 3 };
  })
  .add('Object.freeze (large object)', function () {
    Object.freeze(largeObject);
  })
  .add('Normal assignment (large object)', function () {
    const obj = { ...largeObject };
  })
  .on('cycle', function (event) {
    console.log(String(event.target));
  })
  .on('complete', function () {
    console.log('Fastest is ' + this.filter('fastest').map('name'));
  })
  .run({ async: true });
```